src/pages/ActivityDetails/index.tsx

```diff
@@ -13,11 +15,13 @@ function ActivityDetailsPage() {
   }
 
   return (
-    <Box display="flex" flex={1}>
-      <ActivityDetailsContext.Provider value={{ appletId, activityId, eventId, isPublic: false }}>
-        <ActivityDetailsWidget />
-      </ActivityDetailsContext.Provider>
-    </Box>
+    <AuthorizationGuard fallback={<LoginWithRedirect />}>
```

Comment:
I really think this should be the default `fallback` for AuthorizationGuard, and we should get rid of `AuthorizationButtons`. The interim screens adds zero value to the UX; you can already easily opt whether to log in/sign up from the Login screen. Still might make sense for someone to override the `fallback` prop in theory, so I guess leave that prop available.

Reply:
Agreed. I'll make these changes unless I get pushback about it 👍

Reply:
Actually it occurred to me – it looks like the `AuthorizationButtons` component is passing the `redirectState` to the Signup screen, but I'm not sure if the same thing is happening when we bypass `AuthorizationButtons` and try to navigate to Signup from the Login screen. Can you confirm we're not losing redirect behaviour on Signup?

Reply:
After double checking this, I realise that the login and sign up pages don't pass state to each other. I'm going to revert this change rather than try to resolve this, since it was outside the scope of this PR anyway